### DISCUSSION TITLE
#error: "CUDA atomics are only supported for sm_60 and up on *nix and sm_70 and up on Windows"

### DISCUSSION THREAD

**Discussion by kevkrist** [NVIDIA/cuCollections#425] [Q&A] [answered]
I'm getting this error on a Windows machine with device having compute capability 8.9 (4070Ti). I also get this error on a Linux machine with device having compute capability 6.0 (P100). The error is produced by line 12 of atomic_cuda.h. Any thoughts on this?

**Comment by PointKernel** [comment, marked as answer]
Just FYI, building on Windows (#195) is backlogged and won't be addressed in the short term.

As for the build error itself, you may want to build your code by explicitly specifying a sm_70+ arch, e.g.:
```bash
nvcc -std=c++17 -arch=sm_89 --expt-extended-lambda --expt-relaxed-constexpr mycode.cu
```

**Reply by kevkrist** [reply]
Thank you. Needed to explicitly specify the architecture on Linux.
